core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -341,8 +342,7 @@ public T acquire(long time, TimeUnit unit) throws TimeoutException, IOException
             throw new TimeoutException("Acquire resource times out.");
           }
         } catch (InterruptedException e) {
-          // Restore the interrupt flag so that it can be handled later.
-          Thread.currentThread().interrupt();
+          throw new IOException("Thread interrupted while acquiring client.");
```

Comment:
should we add `this` or some distinguishing identifier so we know which pool this is happening in?

Reply:
Unfortunately we can't get the type of resource without instantiating one. I added this to the message, but that will not be very helpful (just let us know if multiple pools are getting interrupted or just one).

Reply:
Please chain `InterruptedException` to newly created `IOException`?